### PR TITLE
Move static Entity lookup maps to EntityTracker

### PR DESCRIPTION
**Pull request by Minikloon**
EntityTrackerEntry exists to ensure the same behavior as before in regards to the recorded position in the tracker.

### CODE REVIEW

**Comment by mworzala** [comment]
Putting an entity in an instance which already contains that ID or UUID should throw an exception

File: src/main/java/net/minestom/server/instance/EntityTrackerImpl.java [hunk 1]
```diff
@@ -106,17 +138,23 @@ public void remove(@NotNull T removed) {
         }
     }
 
+    @Override
+    public void changeUuid(@NotNull Entity entity, UUID oldUuid) {
```
Comment (mworzala) [inline src/main/java/net/minestom/server/instance/EntityTrackerImpl.java:142]: this is not thread safe and missing an annotation, but honestly we should just remove the setUuuid method from entity and this functionality and make it final. It doesn't make sense to change an entity uuid after creating it I dont think.

**Comment by Minikloon** [comment]
> Putting an entity in an instance which already contains that ID or UUID should throw an exception

It will from the (using Checks) assertion in EntityTracker.

File: src/main/java/net/minestom/server/instance/EntityTrackerImpl.java (hunk 1 above)
Comment (Minikloon) [reply src/main/java/net/minestom/server/instance/EntityTrackerImpl.java:142]: The only use within Minestom of setUuid within Minestom is transitionLoginToConfig
The commit message said thread-safety, I meant safer than register/unregister.

I agree with removing setUwuid, but I think it's beyond the scope of this PR.

**Review by mworzala — APPROVED** [review]

**Pull request merged**